src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -986,6 +986,9 @@ impl LinkCollector<'_, '_> {
                 }
             }
 
+            let stripped_path_string = strip_generic_parameters(path_str);
+            path_str = &stripped_path_string;
```

Comment:
This will never be run because anything with `<` or `>` is skipped up at https://github.com/rust-lang/rust/blob/8a13fc494d4365e57d8f343219f0201458b68591/src/librustdoc/passes/collect_intra_doc_links.rs#L937-L939
You'll need to either move this before, or allow `<` in that bit of code (personally I prefer the second, which would require less changes when https://github.com/rust-lang/rust/issues/74563 gets implemented).